10_integrations/dbt_modal_inference/duck_db_modal_inference.py

```diff
@@ -0,0 +1,276 @@
+# │ Glint        │                2 │               0 │                0 │
+# │ Flumplenook  │                2 │               1 │                1 │
+# │ Whirlybird   │                2 │               0 │                1 │
+# ├──────────────┴──────────────────┴─────────────────┴──────────────────┤
+# │ 11 rows                                                    4 columns │
+# └──────────────────────────────────────────────────────────────────────┘
+# ```
+#
+# Here we can see that the llm classified the results into three different categories
+# that we could then aggregate in a subsequent sql model!
+#
+# Since we're using a volume for storing our dbt target results
+# and our duckdb parquet files
+# you can view the results and use them outside the function too.
+#
+# View the target directory by:
+# ```sh
+# modal volume ls dbt-inference-vol target/
+#            Directory listing of 'target/' in 'dbt-inference-vol'
+# ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━┓
+# ┃ Filename                      ┃ Type ┃ Created/Modified      ┃ Size      ┃
+# ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━┩
+# │ target/run                    │ dir  │ 2024-07-19 22:59 CEST │ 14 B      │
+# │ target/compiled               │ dir  │ 2024-07-19 22:59 CEST │ 14 B      │
+# │ target/semantic_manifest.json │ file │ 2024-07-19 23:25 CEST │ 234 B     │
+# │ target/run_results.json       │ file │ 2024-07-19 23:25 CEST │ 10.1 KiB  │
+# │ target/manifest.json          │ file │ 2024-07-19 23:25 CEST │ 419.7 KiB │
+# │ target/partial_parse.msgpack  │ file │ 2024-07-19 23:25 CEST │ 412.7 KiB │
+# │ target/graph_summary.json     │ file │ 2024-07-19 23:25 CEST │ 1.4 KiB   │
+# │ target/graph.gpickle          │ file │ 2024-07-19 23:25 CEST │ 15.7 KiB  │
+# └───────────────────────────────┴──────┴───────────────────────┴───────────┘
+# ```
+#
+# And the db directory:
+# ```sh
+# modal volume ls dbt-inference-vol db/
+#                   Directory listing of 'db/' in 'dbt-inference-vol'
+# ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━┓
+# ┃ Filename                                 ┃ Type ┃ Created/Modified      ┃ Size    ┃
+# ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━┩
+# │ db/review_sentiments.parquet             │ file │ 2024-07-19 23:25 CEST │ 9.6 KiB │
+# │ db/product_reviews_sentiment_agg.parquet │ file │ 2024-07-19 23:25 CEST │ 756 B   │
+# └──────────────────────────────────────────┴──────┴───────────────────────┴─────────┘
+# ```
+#
+# ## Python dbt model
+# The python dbt model in `dbt_modal_inference_proj/models/product_reviews_sentiment.py` is quite simple.
+# It configures dbt to store the results locally in parquet, fetches the inference url,
+# uses a batch reader to iterate over the inputs and call the `batcher` method
+# and adds a new field from the LLM
+#
+# ```python
```

Comment:
Altogether, there's quite a bit of inlined Python here and we would need to duplicate any changes. Do we need the code here or can we point readers to the files, where we can put comments?

Reply:
That would work, do you think we should link to the files on github directly?